src/Riok.Mapperly/Descriptors/Mappings/NewValueTupleExpressionMapping.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors.Mappings.MemberMappings;
+using Riok.Mapperly.Emit;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using static Riok.Mapperly.Emit.SyntaxFactoryHelper;
+
+namespace Riok.Mapperly.Descriptors.Mappings;
+
+/// <summary>
+/// An object mapping creating the target instance via a tuple expression (eg. (A: 10, B: 20)),
+/// mapping properties via ctor, not by assigning.
+/// <seealso cref="NewInstanceObjectMemberMethodMapping"/>
+/// </summary>
+public class NewValueTupleExpressionMapping : ObjectMemberMethodMapping, INewValueTupleMapping
+{
+    private readonly int _argumentCount;
+    private const string NoMappingComment = "// Could not generate mapping";
+    private const string TargetVariableName = "target";
+    private readonly HashSet<ValueTupleConstructorParameterMapping> _constructorPropertyMappings = new();
+
+    public NewValueTupleExpressionMapping(ITypeSymbol sourceType, ITypeSymbol targetType, int argumentCount)
+        : base(sourceType, targetType)
+    {
+        _argumentCount = argumentCount;
+    }
+
+    public void AddConstructorParameterMapping(ValueTupleConstructorParameterMapping mapping) => _constructorPropertyMappings.Add(mapping);
+
+    public override MethodDeclarationSyntax? BuildMethod(SourceEmitterContext ctx)
+    {
+        // generate method body if complex mapping otherwise return null
+        return HasMemberMappings() ? base.BuildMethod(ctx) : null;
+    }
+
+    public override ExpressionSyntax Build(TypeMappingBuildContext ctx)
+    {
+        // generate error if constructor argument don't match
+        if (_constructorPropertyMappings.Count != _argumentCount)
+        {
+            return ThrowNotImplementedException().WithLeadingTrivia(TriviaList(Comment(NoMappingComment)));
+        }
+
+        // generate method call if it's a complex mapping
+        if (HasMemberMappings())
+        {
+            return base.Build(ctx);
+        }
+
+        // (Name:.. ,..);
+        var ctorArgs = _constructorPropertyMappings.Select(x => x.BuildArgument(ctx, emitFieldName: true)).ToArray();
+        return TupleExpression(SeparatedList(ctorArgs));
```

Comment:
As `SeparatedList` accepts an enumerable as parameter there is no need to collect into an array.